### PR TITLE
Add SimpleMotorBase

### PR DESCRIPTION
**Pull request by csokolove**
Add abstract IntakeBase utilizing SpeedControllerGroups. `forward()`, `backward()`, `stop()` methods included.

### CODE REVIEW

**Comment by csokolove** [comment]
Addresses request for roller intakes in #7

**Review by SteepAtticStairs — APPROVED** [review]

File: subsystems/RollerIntakeBase.java [hunk 1]
```diff
@@ -1,5 +1,70 @@
...
+    public static class IntakeConstants {
+        public boolean inverted;
+        public int stallLimit;
+        public int freeLimit;
+        public double throttle;
+
+        public IntakeConstants(int leaderID, int followerIDs, boolean inverted, int stallLimit, int freeLimit, double throttle) {
+            this.inverted = inverted;
+            this.stallLimit = stallLimit;
+            this.freeLimit = freeLimit;
```
Comment (mxgordon) [inline subsystems/RollerIntakeBase.java]: You shouldn't need all these constants, you only use `throttle` and in that case, you can just pass it in, you don't need `IntakeConstants`

File: subsystems/RollerIntakeBase.java [hunk 2]
```diff
@@ -1,5 +1,70 @@
...
+    protected SpeedController leader;
+    protected SpeedController[] followers;
+    protected SpeedControllerGroup motorGroup;
+
+    // Create intake constants for implementation in methods
+    public static class IntakeConstants {
+        public boolean inverted;
+        public int stallLimit;
+        public int freeLimit;
+        public double throttle;
```
Comment (mxgordon) [inline subsystems/RollerIntakeBase.java]: Removes the unnecessary variables, also make `throttle` protected and not public

File: subsystems/RollerIntakeBase.java [hunk 3]
```diff
@@ -1,5 +1,70 @@
...
+                // put non-leader motor into an array as followers
+                followers[i-1] = motors[i];
+            }
+        }
+
+        // if there are no followers, don't put the followers array in the SpeedControllerGroup
+        if (followers.length == 0) {
+            motorGroup = new SpeedControllerGroup(leader);
+        } else {
+            motorGroup = new SpeedControllerGroup(leader, followers);
```
Comment (mxgordon) [inline subsystems/RollerIntakeBase.java]: You really don't need a leader when you use a `SpeedControllerGroup`, just pass them all in together.

**Review by mxgordon — CHANGES_REQUESTED** [review]

**Comment by csokolove** [comment]
Abstract SimpleMotorBase for use with any motor groups.

File: subsystems/SimpleMotorBase.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public abstract class SimpleMotorBase extends SubsystemBase {
+    protected SpeedControllerGroup motorGroup;
+    protected double throttle;
+
+    public SimpleMotorBase(SpeedController[] motors, double throttle) {
+        motorGroup = new SpeedControllerGroup(motors);
+        this.throttle = throttle;
+    configureMotors();
```
Comment (iscruzrivera) [inline subsystems/SimpleMotorBase.java]: indent this

File: subsystems/SimpleMotorBase.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    public abstract void configureMotors();
+
+    public void forward() {
+        motorGroup.set(throttle);
+    };
```
Comment (iscruzrivera) [inline subsystems/SimpleMotorBase.java]: bestie what

**Review by iscruzrivera — DISMISSED** [review]

File: subsystems/SimpleMotorBase.java (hunk 5 above)
Comment (csokolove) [reply subsystems/SimpleMotorBase.java]: bestie what;

File: subsystems/SimpleMotorBase.java (hunk 4 above)
Comment (csokolove) [reply subsystems/SimpleMotorBase.java]: indent this

**Review by lacedwithennui — APPROVED** [review]
😼

File: subsystems/SimpleMotorBase.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+import edu.wpi.first.wpilibj.SpeedController;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public abstract class SimpleMotorBase extends SubsystemBase {
+    protected SpeedControllerGroup motorGroup;
+    protected double throttle;
+
+    public SimpleMotorBase(SpeedController[] motors, double throttle) {
```
Comment (mxgordon) [inline subsystems/SimpleMotorBase.java:11]: Add some documentation here explaining what this is for, all the other methods are pretty self explainitory

**Review by mxgordon — CHANGES_REQUESTED** [review]

File: subsystems/RollerIntakeBase.java [hunk 7]
```diff
@@ -2,4 +2,4 @@
 
 public abstract class RollerIntakeBase {
```
Comment (mxgordon) [inline subsystems/RollerIntakeBase.java]: You should delete this file

**Review by mxgordon — DISMISSED** [review]

**Pull request merged**